geopy/geocoders/geoclient.py

```diff
@@ -0,0 +1,156 @@
+"""
+:class:`.Geoclient` geocoder.
+"""
+
+from geopy.compat import urlencode
+from geopy.location import Location
+from geopy.util import logger
+from geopy.geocoders.base import Geocoder, DEFAULT_TIMEOUT
+
+from geopy.exc import ConfigurationError
+
+
+__all__ = ("Geoclient", )
+
+
+class Geoclient(Geocoder):
+    """
+    Geocoder using the Geoclient API. Documentation at:
+
+        https://api.cityofnewyork.us/geoclient/v1/doc
+
+     to get access, go to:
+
+        https://developer.cityofnewyork.us/api/geoclient-api   
+    """
+
+    def __init__(
+            self,
+            app_id=None,
+            app_key=None,
+            domain='api.cityofnewyork.us/geoclient/v1',
+            timeout=DEFAULT_TIMEOUT,
+            proxies=None,
+            user_agent=None,
+    ):
+        """
+        Initialize Geoclient geocoder. Please note that 'scheme' parameter is
+        not supported: at present state, all Geoclient traffic use https.
+
+        :param string appd_id: The application ID.
+
+        :param string app_key: The application key. 
+
+        :param string domain: Currently it is 'api.cityofnewyork.us/geoclient/v1', can
+            be changed if you have an on-premise Geoclient instance.
+
+        :param dict proxies: If specified, routes this geocoder's requests
+            through the specified proxy. E.g., {"https": "192.0.2.0"}. For
+            more information, see documentation on
+            :class:`urllib2.ProxyHandler`.
+
+        """
+        super(Geoclient, self).__init__(
+            scheme="https", timeout=timeout, proxies=proxies, user_agent=user_agent
+        )
+
+        self.app_id = app_id
+        self.app_key = app_key
+        
+        if not app_id:
```

Comment:
Actually, `app_id` and `app_key` are only required when using the `DEFAULT_DOMAIN`.  

If using an on-premise installation, we do not need to pass thru the 3Scale API keys used by the City for its managed Geoclient instance.